app/models/hotel.rb

```diff
@@ -1,3 +1,9 @@
 class Hotel < ApplicationRecord
-   has_many :rooms
+  has_many :rooms
+
+  validates_presence_of :name, :location
+
+  def unique_guests
+    Hotel.joins(rooms: :guests).where(id: self.id).distinct.pluck("guests.name")
```

Comment:
Good job with this join! You can make this a bit easier to find by adding an association to this `Hotel` model: 
```
has_many :guests, through: :rooms
```
Then, you could just call `guests.distinct.pluck(:name)`.